# Anforderungen an Vorbeschleunigerstruktur

Nachfolgend aufgeführt sind die Anforderungen, die der Hauptbeschleuniger für die $t\bar{t}$-Produktion an die Vorbeschleuniger und die Teilchenquelle stellen. Dabei erfolgt eine Orientierung an der LHC-Vorbeschleunigerstruktur. Insbesonder bzgl. der Einschuss-/Extraktionsenergie der einzelnen Vorbeschleuniger. Diese wird u.a. durch Restmagnetisierung der Dipole etc. bestimmt.

**Einschussenergie: 130 GeV  
Einschussstrom: 0,5 A**

Betrachte die Vorbeschleunigerstruktur rückwärts, also beginnend beim letzten Vorbeschleuniger. Dieser muss die oben angegebene Einschussenergie von 130 GeV und einen Einschussstrom von 0,5 A liefern.

In [93]:
import numpy as np
from scipy import constants 

In [94]:
# Eingehende Groeßen von anderen Notebooks

n_b = 4000  # Bunchanzahl pro Strahl im Hauptbeschleuniger
N = 0.16*10**(11) # Anzahl Teilchen im Bunch

# Vorgegeben
Einschussenergie = 130 # GeV

**Weiter unten befinden sich allgemeine Rechnungen, um die Vorbeschleunigerstruktur nach und nach zu erarbeiten. Der an diese Zelle angeschlossen Teil stellt die bisherige tatsächlich vorgeschlagene Vorbeschleunigerstruktur dar.**

### Die Protonenquelle  

Wasserstoffatomen wird durch e$^-$-Beschuss ihr eigenes Elektron entrissen. Die positiv geladenen Ionen (H$^+$) werden dann in die (Vor-)Bescleunigerstruktur extrahiert. Das Extrhieren führt bereits zu einer ersten (gerichteten) Beschleunigung der Ionen. Hier also der Protonen.  

Evtl. Perveanz, Emittanz, Brillianz einbringen.
Strom, Teilchenanzahl, etc. berechnen

Bild einer Ionenquelle einfügen.

### Vorbeschleuniger 4 (nachfolgend Einspeisung in Hauptbeschl.): Synchrotron  

$E_{max} \approx recB$,  

mit der Maximalenergie $E_{max}$ des Synchrotrons, dem Radius r, der Elektronladung e, der Lichtgeschwindigkeit c und der Magnetfeldstärke B. Der Umfang des Vorbeschleunigers sollte deutlich unter dem des Hauptbeschleunigers liegen. Auch die Magnete sollten geringere Feldstärken stellen müssen. -> Für B verschiedene Annahmen machen und schauen, welcher Radius bzw. Umfang sich daraus ergibt. Beachte gleichzeitig die Bunchanzahl (eingehend über harmonische Zahl h), um genügend Strahlstrom zu liefern: $$f_{RF} = hf_{rev}.$$

Aufstellung Magnetfeldstärke, sich ergebender Radius und resultierender Umfang.

| B[T] |  r[m]  |   U[m]  |
|:----:|:------:|:-------:|
|  0,5 | 867,26 | 5449,16 |
|  1,0 | 466,53 | 2724,58 |
|  1,5 | 289,06 | 1816,22 |
|  2,0 | 216,81 | 1362,26 |
|  2,5 | 173,45 | 1089,82 |
|  3,0 | 144,54 |  908,17 |
|  3,5 | 123,89 |  778,42 |
|  4,0 |  108,4 |  681,1  |
|  4,5 | 96,362 |  605,46 |
|  5,0 | 86,726 |  544,92 |

Bis ca. 2 T kann mit konventionellen Elektromagneten gearbeitet werden, SPS am CERN nutzt auch konventionelle Magnete. Da dies deutlich einfacher zu bauen und zu betreiben ist und gleichzeitig höhere Magnetfeldstärken verhältnismäßig wenig zu geringerem Umfang beitragen, sollte hier B = 1,5T oder B = 2,0 T gewählt werden.  -> weitere Entscheidungskriterien?

In [65]:
# Pythoncode fuer obige Tabelle
B_list = [0.5, 1.0, 1.5, 2.0, 2.5, 3.0, 3.5, 4.0, 4.5, 5.0]
Radien = []
for i in range(len(B_list)):
    r = 130*10**9/(1*constants.c*B_list[i])
    Radien.append(r)

#print("Radius [m]:", Radien)

Umfang = [] 
for k in range(len(Radien)):
    U = 2*constants.pi*Radien[k]
    Umfang.append(U)

#print("\nUmfang [m]:", Umfang)

In [3]:
# Potentielle Magnetfeldstärken im Vorbeschleuniger 
B1 = 1.5  # Tesla
B2 = 2.0  # Tesl

CERN: Bei Injektion in den letzten Vorbeschleuniger (SPS) haben die Protonen  ca. 25 GeV -> v $\approx$ c. Es folgt eine Umlauffrequenz

In [66]:
# Potentieller Umfang des Vorbeschleunigers
U1 = 1816.22 # in Metern, gilt fuer B1 = 1,5 T
U2 = 1362.26 # in Metern, gilt fuer B1 = 2,0 T

In [95]:
# Umlauffrequenz: f_rev = v/s, für v ist c anzunehmen und für die Strecke s der aus obiger Tabelle entnommene Umfang des Synchrotrons

f_revB1 = (constants.c/U1)/10**3    # in kHz gerechnet
f_revB2 = (constants.c/U2)/10**3

print("Fuer den Vorbeschleuniger 4 ergibt sich eine Umlauffrequenz von:")
print("Bei 1,5T:",f_revB1,"kHz", "\nBei 2,0T:", f_revB2, "kHz")

Fuer den Vorbeschleuniger 4 ergibt sich eine Umlauffrequenz von:
Bei 1,5T: 165.06395590842519 kHz 
Bei 2,0T: 220.06992644575925 kHz


Bei 1,5T -> 1816,22m Umfang -> 165,06kHz  
Bei 2,0T -> 1362,26m Umfang -> 220,07kHz

Um den Hauptbeschleuniger mit einer Füllung des letzten Vorbeschleunigers zu füllen, werden also 2x4000 = 8000 Bunche mit je 0,16*$10^{11}$ Teilchen benötigt. Für diese Anzahl Bunche benötigt dieser Vorbeschleuniger eine HF von:

In [96]:
# Unter Betrachtung der gewuenschten Bucketanzahl im Beschleuniger ergeben sich die dafür benötigten RFs 
f_RFEinzelfuellung1 = 8000 * 165.06
f_RFEinzelfuellung2 = 8000 * 220.07
f_RF1 = 2000 * 165.06 
f_RF2 = 2000 * 220.07
f_RF3 = 1000 * 165.06
f_RF4 = 1000 * 220.07

In [97]:
print(f_RFEinzelfuellung1/10**3, "MHz")
print(f_RFEinzelfuellung2/10**3, "MHz")
#print(f_RF1, "kHz") 
print(f_RF1/10**3, "MHz")
#print(f_RF2, "kHz") 
print(f_RF2/10**3, "MHz")
#print(f_RF3, "kHz") 
print(f_RF3/10**3, "MHz")
#print(f_RF4, "kHz") 
print(f_RF4/10**3, "MHz")

1320.48 MHz
1760.56 MHz
330.12 MHz
440.14 MHz
165.06 MHz
220.07 MHz


Wähle darauf basierend die Bunchanzahl und den Beschleunigerumfang (durch f_rev)

In [98]:
# Fuer 2000 Buckets und 165.06 Hz f_rev
print("Fuer 2000 Buckets und 165.06 Hz f_rev (1816,22m Umfang):", f_RF1/10**3, "MHz")

#Fuer 2000 Buckets und 220.07 Hz f_rev
print("Fuer 2000 Buckets und 220.07 Hz f_rev (1362,26m Umfang):", f_RF2/10**3, "MHz")

Fuer 2000 Buckets und 165.06 Hz f_rev (1816,22m Umfang): 330.12 MHz
Fuer 2000 Buckets und 220.07 Hz f_rev (1362,26m Umfang): 440.14 MHz


In [104]:
# Strahlstrom fuer alle Buckets gefuellt und fuer die Haelfte der Buckets gefuellt
Strahlstrom1 = 2000*0.16*10**(11)*165.06*10**(3)*1.6*10**(-19) #in Ampere
Strahlstrom2 = 1000*0.16*10**(11)*165.06*10**(3)*1.6*10**(-19) #in Ampere
Strahlstrom3 = 2000*0.16*10**(11)*220.07*10**(3)*1.6*10**(-19) #in Ampere
Strahlstrom4 = 1000*0.16*10**(11)*220.07*10**(3)*1.6*10**(-19) #in Ampere

In [105]:
print("Strahlstrom je nach Umlauffrequenz:")
print("2000 Bunche", 165.06, "kHz:", Strahlstrom1, "A")
print("1000 Bunche", 165.06, "kHz:", Strahlstrom2, "A")
print("2000 Bunche",220.0, "kHz:", Strahlstrom3, "A")
print("1000 Bunche",220.0, "kHz:", Strahlstrom4, "A")

Strahlstrom je nach Umlauffrequenz:
2000 Bunche 165.06 kHz: 0.8451072 A
1000 Bunche 165.06 kHz: 0.4225536 A
2000 Bunche 220.0 kHz: 1.1267584 A
1000 Bunche 220.0 kHz: 0.5633792 A


**Für den Vorbeschleuniger 4 (Injektor in den Hauptbeschleuniger) wird folgende Konfiguration gewählt:**
- Magnete: konventionell, 2T
- Umfang: 1362,26m
- Buckets: 2000
- Bunche: 1000
- Umlauffrequenz: 220.07kHz
- RF: 440.14 MHz
- Füllungen für den Hauptbeschleuniger: 8
- Strahlstrom: 0,56 A
- Beschleunigung: 25GeV -> 130GeV (orientiert an CERN)

Basierend auf den bisherigen Ergebnissen benötigt der Hauptbeschleuniger 8 Füllungen des Vorbeschleunigers 4.Bestimmend, welche Injektionsenergie benötigt wird, ist insbesondere das Magnetfeld des Beschleunigers. Aufgrund des Restmagnetfeldes im Joch muss eine gewisse Injektionsenergie bestehen, um die Teilchen auf einer stabilen Bahn zu halten. Am CERN kommt vor dem SPS das Proton-Synchrotron (PS). PS versorgt SPS mit Protonen der Energie 25 GeV -> dies soll auch hier der angenommene Wert sein, den der letzte Vorbeschleuniger noch benötigt.

Bestimmend, welche Injektionsenergie benötigt wird, ist insbesondere das Magnetfeld des Beschleunigers. Aufgrund des Restmagnetfeldes im Joch muss eine gewisse Injektionsenergie bestehen, um die Teilchen auf einer stabilen Bahn zu halten. Am CERN kommt vor dem SPS das Proton-Synchrotron (PS). PS versorgt SPS mit Protonen der Energie 25 GeV -> dies soll auch hier der angenommene Wert sein, den der letzte Vorbeschleuniger noch benötigt.

### Vorbeschleuniger 3: Synchrotron

Gleiche Überlegungen wie zuvor:  


| B[T] |  r[m]  |   U[m]  |
|:----:|:------:|:-------:|
|  0,5 | 166,78 | 1047,91 |
|  1,0 | 83,391 |  523.96 |
|  1,5 | 55,594 |  349,31 |
|  2,0 | 41,695 |  261,98 |
|  2,5 | 33,356 |  208,95 |
|  3,0 | 27,797 |  174,65 |
|  3,5 | 23,826 |  149,70 |
|  4,0 | 20,847 |  130,99 |
|  4,5 | 18,531 |  116,43 |
|  5,0 | 16,678 |  104,79 |


In [106]:
BS1 = 2.0  # Tesla
BS2 = 1.5  # Tesla
BS3 = 1.0  # Tesla
BS4 = 0.5  # Tesla

In [107]:
# Potentieller Umfang des Vorbeschleunigers
US1 = 1047.91 # in Metern, gilt fuer B1 = 0,5 T
US2 = 523.96 # in Metern, gilt fuer B1 = 1,0 T
US3 = 349.31 # in Metern, gilt fuer B1 = 1,5 T
US4 = 261.98 # in Metern, gilt fuer B1 = 2,0 T

In [108]:
# Umlauffrequenz: f_rev = v/s, für v ist c anzunehmen und für die Strecke s der aus obiger Tabelle entnommene Umfang des Synchrotrons

f_revBS1 = (constants.c/US1)/10**3
f_revBS2 = (constants.c/US2)/10**3
f_revBS3 = (constants.c/US3)/10**3
f_revBS4 = (constants.c/US4)/10**3

print("Bei 0,5T:",f_revBS1,"kHz", "Bei 1,0 T:", f_revBS2, "kHz", "Bei 1,5 T:", f_revBS3,"kHz", "Bei 2,0 T:", f_revBS4, "kHz")

Bei 0,5T: 286.08607418576025 kHz Bei 1,0 T: 572.1666882968166 kHz Bei 1,5 T: 858.2418424894793 kHz Bei 2,0 T: 1144.3333765936331 kHz


In [109]:
f_RFS1 = 2000 * 858.242 # Für 2000 Buckets 
f_RFS2 = 2000 * 1144.333
f_RFS3 = 2000 * 572.167
f_RFS4 = 2000 * 286.086
f_RFS5 = 1000 * 858.242 # Für 1000 Buckets 
f_RFS6 = 1000 * 1144.333
f_RFS7 = 1000 * 572.167
f_RFS8 = 1000 * 286.086

In [119]:
#print(f_RFS1, "kHz") 
print("2000 Buckets")
print("Umlauffrequenz, RF")
print("286.08 kHz:", f_RFS1/10**3, "MHz")
#print(f_RFS2, "kHz") 
print("1144.333 kHz:" ,f_RFS2/10**3, "MHz")
#print(f_RFS3, "kHz") 
print("572.167 kHz:", f_RFS3/10**3, "MHz")
#print(f_RFS4, "kHz") 
print("286.086 kHz:", f_RFS4/10**3, "MHz")
print("\n1000 Buckets")
print("Umlauffrequenz, RF")
#print(f_RFS5, "kHz") 
print("858.242 kHz:", f_RFS5/10**3, "MHz")
#print(f_RFS6, "kHz") 
print("1144.333 kHz:", f_RFS6/10**3, "MHz")
#print(f_RFS7, "kHz") 
print("572.167 kHz:", f_RFS7/10**3, "MHz")
#print(f_RFS8, "kHz") 
print("286.086 kHz:", f_RFS8/10**3, "MHz")

2000 Buckets
Umlauffrequenz, RF
286.08 kHz: 1716.484 MHz
1144.333 kHz: 2288.666 MHz
572.167 kHz: 1144.334 MHz
286.086 kHz: 572.172 MHz

1000 Buckets
Umlauffrequenz, RF
858.242 kHz: 858.242 MHz
1144.333 kHz: 1144.333 MHz
572.167 kHz: 572.167 MHz
286.086 kHz: 286.086 MHz


In [115]:
Strahlstrom_S1 = 2000*0.16*10**(11)*286.086*10**(3)*1.6*10**(-19) #in Ampere
Strahlstrom_S2 = 1000*0.16*10**(11)*286.086*10**(3)*1.6*10**(-19) #in Ampere
Strahlstrom_S3 = 500*0.16*10**(11)*286.086*10**(3)*1.6*10**(-19) #in Ampere

In [117]:
print("2000 Bunche, Umlauffrequenz 286,086", Strahlstrom_S1, "A", "\n1000 Bunche, Umlauffrequenz 286,086", Strahlstrom_S2, "A", "\n500 Bunche, Umlauffrequenz 286,086", Strahlstrom_S3, "A")

2000 Bunche, Umlauffrequenz 286,086 1.46476032 A 
1000 Bunche, Umlauffrequenz 286,086 0.73238016 A 
500 Bunche, Umlauffrequenz 286,086 0.36619008 A


**Für den Vorbeschleuniger 3 wird folgende Konfiguration gewählt:**
- Magnete: konventionell, 0,5T
- Umfang: 1047.91m
- Buckets: 1000
- Bunche: 500
- Umlauffrequenz: 286.09kHz
- RF: 286,086 MHz
- Füllungen für den Hauptbeschleuniger: 16
- Strahlstrom: 0,37 A
- Beschleunigung: 1,4GeV -> 25GeV (orientiert an CERN)

**Nachfolgend die zur Erarbeitung genutzten Code-Bausteine und Rechnungen**

### Vorbeschleuniger 2

Das CERN PS wird mit 1,4 GeV Protonen aus dem Protonen Booster versorgt. Entsprechend soll auch hier die Vorbeschleunigerstufe vor dem vorletzten Beschleuniger Protonen mit 1,4 GeV bereitstellen. ->Mehrere Synchrotronstrahlrohre übereinander.  
#### Ab hier: nicht mehr mit c als Umlauffrequenz rechnen. Bei 1,4 GeV gilt v $\approx$ 0,74*c = 221846418.92 m/s. Für Ringumfang wieder obige Formel verwendet.

| B[T] | r[m] | U[m]  |
|------|------|-------|
| 0,5  | 9,34 | 58,68 |
| 1,0  | 4,67 | 29,34 |
| 1,5  | 3,11 | 19,54 |
| 2,0  | 2,33 | 14,64 |
| 2,5  | 1,87 | 11,75 |

In [121]:
# Umlauffrequenz: f_rev = v/s, für v ist 0,74*c anzunehmen und für die Strecke s der aus obiger Tabelle entnommene Umfang des Synchrotrons

f_revB1 = (0.74*constants.c/58.68)/10**3    # in kHz gerechnet
f_revB2 = (0.74*constants.c/29.34)/10**3
f_revB3 = (0.74*constants.c/19.54)/10**3
f_revB4 = (0.74*constants.c/14.64)/10**3
print("Bei 0,5T:",f_revB1,"kHz", "\nBei 1,0T:", f_revB2, "kHz","\nBei 1,5T:",f_revB3,"kHz", "\nBei 2,0T:", f_revB4, "kHz")

Bei 0,5T: 3780.6138193592365 kHz 
Bei 1,0T: 7561.227638718473 kHz 
Bei 1,5T: 11353.45030296827 kHz 
Bei 2,0T: 15153.443915300544 kHz


In [132]:
f_RF1 = 2000 * 3780.61 # Für 2000 Bunche je Füllung werden insgesamt ca. vier Füllungen des Hauptbeschleunigers benötigt
f_RF2 = 2000 * 7561.23
f_RF3 = 2000 * 11353.45 # Für 2000 Bunche je Füllung werden insgesamt ca. vier Füllungen des Hauptbeschleunigers benötigt
f_RF4 = 2000 * 15153.44
f_RF5 = 25 * 3780.61
f_RF6 = 25 * 7561.23
f_RF7 = 25 * 11353.45 # Für 2000 Bunche je Füllung werden insgesamt ca. vier Füllungen des Hauptbeschleunigers benötigt
f_RF8 = 25 * 15153.44
f_RF_1 = 2 * 3780.61

In [133]:
print(f_RF1, "kHz") 
print(f_RF1/10**3, "MHz")
print(f_RF2, "kHz") 
print(f_RF2/10**3, "MHz")
print(f_RF3, "kHz") 
print(f_RF3/10**3, "MHz")
print(f_RF4, "kHz") 
print(f_RF4/10**3, "MHz")
print(f_RF5, "kHz") 
print(f_RF5/10**3, "MHz")
print(f_RF6, "kHz") 
print(f_RF6/10**3, "MHz")
print(f_RF7, "kHz") 
print(f_RF7/10**3, "MHz")
print(f_RF8, "kHz") 
print(f_RF8/10**3, "MHz")
print(f_RF_1/10**3, "MHz")

7561220.0 kHz
7561.22 MHz
15122460.0 kHz
15122.46 MHz
22706900.0 kHz
22706.9 MHz
30306880.0 kHz
30306.88 MHz
94515.25 kHz
94.51525 MHz
189030.75 kHz
189.03075 MHz
283836.25 kHz
283.83625 MHz
378836.0 kHz
378.836 MHz
7.5612200000000005 MHz


In [134]:
Strahlstrom1 = 25*0.16*10**(11)*3780.61*10**(3)*1.6*10**(-19) #in Ampere
Strahlstrom2 = 25*0.16*10**(11)*7561.23*10**(3)*1.6*10**(-19) #in Ampere
Strahlstrom3 = 25*0.16*10**(11)*11353.45*10**(3)*1.6*10**(-19) #in Ampere
Strahlstrom4 = 25*0.16*10**(11)*15153.44*10**(3)*1.6*10**(-19) #in Ampere
Strahlstrom5 = 1*0.16*10**(11)*3780.61*10**(3)*1.6*10**(-19) #in Ampere

In [135]:
print(Strahlstrom1, "A")
print(Strahlstrom2, "A")
print(Strahlstrom3, "A")
print(Strahlstrom4, "A")
print(Strahlstrom5, "A")

0.24195904 A
0.48391871999999997 A
0.7266208 A
0.9698201599999999 A
0.0096783616 A


In [136]:
# Für vier Strahlrohre wie beim Protonen Booster am LHC
print(4*Strahlstrom1, "A")
print(4*Strahlstrom2, "A")
print(4*Strahlstrom3, "A")
print(4*Strahlstrom4, "A")
print(4*Strahlstrom5, "A")

0.96783616 A
1.9356748799999999 A
2.9064832 A
3.8792806399999997 A
0.0387134464 A


**Für den Vorbeschleuniger 2 wird folgende Konfiguration gewählt:**
- Magnete: konventionell, 0,5T
- Umfang: 58.68m
- Buckets: 2 pro Strahlrohr
- Bunche: 1 pro Strahlrohr
- 4 Strahlrohre
- Umlauffrequenz: 3780.61kHz
- RF: 7.56 MHz
- Füllungen für den Hauptbeschleuniger: 2000
- Strahlstrom: 0,0097 A pro Rohr, 0,039 A für 4 Rohre
- Beschleunigung: 50MeV -> 1,4GeV (orientiert an CERN)


### Erster Vorbeschleuniger: Linac  
Orientiert amLinac 2 am CERN

Protonenquelle (CERN) liefert Protonen mit 90 keV, der Linac 2 beschleunigt sie auf 50 MeV. Zur Berechnung wird wie in Übungsblatt 2 vorgegangen: Beschleunigungsstrecke -> 75 mm, Beschleunigungsspannung 100 kV (hier ggf. angepasst), Frequenz 200MHz, Beschleunigung auf 0,314 % der Lichtgeschwindigkeit, Start bei 90 keV (gem. CERN)


In [137]:
import scipy.constants as sci

# Fuer die Fallstudie

# Ruhenergie Proton in MeV
m = sci.physical_constants["proton mass energy equivalent in MeV"]
#print(m[0])
# Lichtgeschwindigkeit
c = sci.c
#print(c)


# Laenge einer Driftroehre, Fkt. von Beta-Faktor aus separater Fkt. abh.
# Formel aus Vorlesung, c/200000000 isst Wellenlaenge der RF
def LDrift(b):
    L = 0.5*(c/200000000)*b
    return L
 
# Beta-Faktor in Abh. der Ruheenergie und der n-fach erhoehten Ekin,
#Erhoehung bei jedem Durchgang durch die Beschleunigungsstrecke
def Beta(n, m):
    B = (1 - ((m*(10**6))**2/(((90*(10**3) + n*200*(10**3)) + m*(10**6))**2)))**(0.5)
    return B


# Fuer Protonen
print("Fuer Protonen")
# Anzahl noetiger Driftroehren: Liegt Beta bei ungefaehr 1, kann Laenge der
# Roehren gleich bleiben
Zaehler = 0
n = 1 # Annahme, dass zuerst eine Beschleunigung durchgefuehrt wird
beta = Beta(n, m[0]) # Startwert fuer Beta, nach Vorbeschl. u. erste Beschl.
#print(beta)

# Schleife laueft, bis Beta zum ersten mal knapp an 0,314 liegt (Geschw. bei 50MeV), 
# solange wird
# mit dem n die Anzahl Beschleunigungsdrchgaenge gezaehlt und mit Zaehler
# die Anzahl durchlaufener Driftroehren
while beta < 0.314:
    Zaehler = Zaehler + 1
    n = n + 1
    beta = Beta(n, m[0])
    
print("Anzahl Beschleunigungen:", Zaehler)  

# Laenge der Roehren kann ab Zaehler (nach w-Schleife) plus 1 konstant gehalten
# werden, plus 1 ist die dann konstant haltbare Roehrenlaenge  
    
#print(LDrift(0.9))
 
Rohr = 0
n = 1

# Driftroehrenlaengen berechnen und Addieren
for i in range(Zaehler + 1):
    DL = LDrift(Beta(n, m[0]))
    Rohr = Rohr + DL
    n = n + 1
    #if i > 57120: print(i, DL, Rohr)
    #elif i< 2: print(i, DL, Rohr)
 
#print(Rohr)
#print(DL)
# Laenge des Beschleunigers bis Driftroehrenlaenge konstant sein kann, hier ist
# die entsprechend oft durchlaufene Beschleunigungsstrecke einzurechnen
print("Der Beschleuniger benoetigt eine Laenge von", Rohr + Zaehler*0.075, "m")   
  

Fuer Protonen
Anzahl Beschleunigungen: 249
Der Beschleuniger benoetigt eine Laenge von 58.7257138038766 m


**Für den Vorbeschleuniger 1 wird folgende Konfiguration gewählt:**
- RF: 200 MHz
- Länge: 58,73m
- Füllungen für den Hauptbeschleuniger: 8000
- Beschleunigung: 90keV -> 50MeV (orientiert an CERN)